src/main/java/by/itech/lab/supplier/service/impl/CategoryService.java

```diff
@@ -0,0 +1,67 @@
+package by.itech.lab.supplier.service.impl;
+
+import by.itech.lab.supplier.domain.Category;
+import by.itech.lab.supplier.dto.CategoryDto;
+import by.itech.lab.supplier.dto.mapper.CategoryMapper;
+import by.itech.lab.supplier.repository.CategoryRepository;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+public class CategoryService {
+
+    private final CategoryRepository categoryRepository;
+
+    private final CategoryMapper mapper;
+
+    public CategoryService(final CategoryRepository categoryRepository,
+                           final CategoryMapper mapper) {
+        this.categoryRepository = categoryRepository;
+        this.mapper = mapper;
+    }
+
+    public CategoryDto findByCategory(final String categoryName) {
+        return categoryRepository.findByCategory(categoryName)
+          .map(mapper::map).orElse(null);
+    }
+
+    public CategoryDto save(final CategoryDto dto) {
+        Category category;
+        if (Objects.isNull(dto.getId())) {
+            category = mapper.map(dto);
+        } else {
+            category = mapper.mapCategoryWithId(dto);
+        }
+        CategoryDto created = mapper.map(categoryRepository.save(category));
+        return created;
```

Comment:
why its separate line? are there some requirements for this?

Reply:
done